Encounter Me/Pages/Index.razor

```diff
@@ -1,26 +1,115 @@
 ﻿@page "/"
 @inject IJSRuntime JSRuntime
 @inject HttpClient Http
+@inject NavigationManager NavManager
 @using ExtensionMethods
+@using BrowserInterop.Extensions
+@using BrowserInterop.Geolocation
+@implements IAsyncDisposable
+
+<div id="map"></div>
+<div class="modal fade @alertModal.modalClass" tabindex="-1" role="dialog" style="display:@alertModal.modalDisplay; overflow-y: auto;" id="alertModal">
+    <div class="d-flex justify-content-center">
+        <div class="modal-dialog modal-confirm">
+            <div class="modal-content">
+                    <div class="modal-header-warning justify-content-center">
+                        <img src="Images/alert.png">
+                    </div>
+                    <div class="modal-body text-center">
+                        <h4>Alert!</h4>
+                        <h5>It is recommended to enable location on your browser for trail browsing.</h5>
+                        <button class="btn btn-success" @onclick="(e => { ModalControl.CloseModal(alertModal);  StateHasChanged();})" data-dismiss="modal"><span>Understood</span></button>
+                    </div>
+            </div>
+        </div>
+    </div>
+</div>
+
+<div class="modal fade @errorModal.modalClass" tabindex="-1" role="dialog" style="display:@errorModal.modalDisplay; overflow-y: auto;" id="errorModal">
+    <div class="d-flex justify-content-center">
+        <div class="modal-dialog modal-confirm">
+            <div class="modal-content">
+                <div class="modal-header-error justify-content-center">
+                    <img src="Images/alert.png">
+                </div>
+                <div class="modal-body text-center">
+                    <h4>Error!</h4>
+                    <h5>Trail walking is supported on mobile devices only!</h5>
+                    <button class="btn btn-success" @onclick="(e => { ModalControl.CloseModal(errorModal); StateHasChanged(); })" data-dismiss="modal"><span>Understood</span></button>
+                    <button class="btn btn-cancel" @onclick="(e => TestTrailLoad())" data-dismiss="modal"> <span>Skip warning (development only)</span></button>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
 
-<div id="mapid"></div>
 
 @code{
 
     List<TrailContainer> trails = new List<TrailContainer>();
 
+    private WindowNavigatorGeolocation geolocationWrapper;
+    private GeolocationPosition geolocationPosition;
+
+    private List<GeolocationPosition> positionHistory = new List<GeolocationPosition>();
+
+    private IAsyncDisposable geopositionWatcher;
+
+    private Modal alertModal = new Modal();
+    private Modal errorModal = new Modal();
+
```

Comment:
Continuing to nit-pick on properties naming conventions. I think we could do a separate branch to clean up the code latter, but we should discuss how we want to name private properties. In my code usually  I start them with underscore like this: _privatePropertyNameHere, I saw that Dominykas did the same. Once again - this feature already has a lot of changes, so I think it would be enough to add code clean up to out todo-list and finish it another day.

Reply:
updated field naming in newest commit